### DISCUSSION TITLE
Where should `linkaxes!` go?

### DISCUSSION THREAD

**Discussion by Klafyvel** [JuliaPlots/Makie.jl#3555] [Q&A] [answered]
Hi there! I found myself making faulty plots several times lately because axes I thought I had linked ended up not being linked. Is there a specific place where `linkaxes!` should be positioned? In the documentation it's placed right after the definition of the axes, but if I do this:

```julia
set_theme!()
f = Figure()
ax = Axis(
    f[1, 1];
    xlabel="Energy [eV]",
    ylabel="PL [norm.]",
)
ax_abs = Axis(
    f[1, 1];
    ylabel="Absorption [norm.]",
    yaxisposition=:right,
)
linkaxes!(ax, ax_abs)
hidespines!(ax_abs)

energies = range(start=1, stop=3, length=500)
abso = @. exp(-(energies - 2)^2 / (2*0.01)^2)
lines!(ax, energies, abso)
lines!(ax_abs, energies, abso, color=Cycled(2))
f
save("link_before_plotting.png", f)
``` 

I get the following plot:
![bug_makie](https://github.com/MakieOrg/Makie.jl/assets/6209071/f2aaaa85-ac77-4a3d-a884-ab68ab919b95)

Whereas if I link after plotting:
```julia
set_theme!()
f = Figure()
ax = Axis(
    f[1, 1];
    xlabel="Energy [eV]",
    ylabel="PL [norm.]",
)
ax_abs = Axis(
    f[1, 1];
    ylabel="Absorption [norm.]",
    yaxisposition=:right,
)
hidespines!(ax_abs)

energies = range(start=1, stop=3, length=500)
abso = @. exp(-(energies - 2)^2 / (2*0.01)^2)
lines!(ax, energies, abso)
lines!(ax_abs, energies, abso, color=Cycled(2))
linkaxes!(ax, ax_abs)
f
save("link_after_plotting.png", f)
```

I get the correct plot:
![bug_makie](https://github.com/MakieOrg/Makie.jl/assets/6209071/77bc74c8-f448-460c-8c61-56701447e761)

Is this a bug, or do I misunderstand the goal of `linkaxes!`?

**Comment by jkrumbiegel** [comment, marked as answer]
This must be a bug, linked axes should stay synchronized if you keep plotting stuff.

**Reply by Klafyvel** [reply]
Ok, thanks. Do you want me to copy this question into an issue?

**Reply by jkrumbiegel** [reply]
Yes that's a good idea. The update order with limits is a bit tricky so it would be good to write more tests for it, too.
